src/components/inputs/FileUploader.tsx

```diff
@@ -0,0 +1,152 @@
+import { CloseRounded, FileUploadRounded, GridViewRounded, ListRounded } from '@mui/icons-material';
+import { Box, Grid, IconButton, List, ListItem, ListItemIcon, ListItemText, Paper, Stack, Typography } from '@mui/material';
+import { makeStyles } from 'makeStyles';
+import { useCallback, useEffect, useRef, useState } from 'react';
+import { useDropzone } from 'react-dropzone';
+
+const useStyles = makeStyles()((theme) => ({
+  fileUploader: {
+    display: 'block',
+  },
+  uploadBox: {
+    padding: theme.spacing(2),
+    borderRadius: `${theme.shape.borderRadius}px`,
+    border: '2px solid white',
+    '&:hover': {
+      backgroundColor: theme.palette.primary.main,
+      opacity: [0.9, 0.8, 0.7],
+    },
+  },
+}));
+
+type GridFileItemProps = {
+  index: number;
+  deleteFile: (index: number) => void;
+  file: File;
+};
+
+const GridFileItem = ({ index, deleteFile, file }: GridFileItemProps) => {
+  const ImageURL = useRef(URL.createObjectURL(file));
+  const isImage = file.type === 'image/jpeg' || file.type === 'image/png';
+
+  useEffect(() => {
+    return URL.revokeObjectURL(ImageURL.current);
+  }, []);
+
+  return (
+    <Grid item key={index} md={4} sm={4} xs={2}>
+      <Paper
+        elevation={2}
+        sx={{
+          height: '100px',
+          background: isImage ? `url(${ImageURL.current})` : 'white',
+          backgroundPosition: 'center',
+          backgroundRepeat: 'no-repeat',
+          backgroundSize: 'contain',
+        }}>
+        <IconButton
+          onClick={() => deleteFile(index)}
+          size='small'
+          sx={{
+            background: 'rgba(0,0,0,0.2)',
+            '&:hover': {
+              background: 'rgba(0,0,0,0.5)',
+            },
+          }}>
+          <CloseRounded />
+        </IconButton>
+
+        {!isImage && (
+          <Box sx={{ px: 2 }}>
+            <Typography color='black' fontWeight='bold' noWrap={true} textAlign='center'>
+              {file.name}
+            </Typography>
+            <Typography color='black' textAlign='center'>
+              {file.type.split('/')[1]}
+            </Typography>
+          </Box>
+        )}
+      </Paper>
+    </Grid>
+  );
+};
+
+type FileUploaderProps = {
```

Comment:
```suggestion
export type FileUploaderProps = {
```

Eksporter props når komponenten som høres til exporteres